common/src/lru.rs

```diff
@@ -1,925 +1,286 @@
-// Taken from https://raw.githubusercontent.com/jeromefroe/lru-rs/master/src/lib.rs @ a11d47ddc7ab86ff07e2308e96fa0a03bcc3f385
-// We copied this into our repository because for no_std builds we need the hashbrown feature, but
-// without ahash. Without introducing all of hashbrown's features into the lru crate, there is no
-// way to include hashbrown in our desired configuration.
-// The code was modified to always enable the hashbrown/nightly features.
-
-// MIT License
-
-// Copyright (c) 2016 Jerome Froelich
-
-// Permission is hereby granted, free of charge, to any person obtaining a copy
-// of this software and associated documentation files (the "Software"), to deal
-// in the Software without restriction, including without limitation the rights
-// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
-// copies of the Software, and to permit persons to whom the Software is
-// furnished to do so, subject to the following conditions:
-
-// The above copyright notice and this permission notice shall be included in all
-// copies or substantial portions of the Software.
-
-// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-// FITNpubESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
-// SOFTWARE.
-
-//! An implementation of a LRU cache. The cache supports `get`, `get_mut`, `put`,
-//! and `pop` operations, all of which are O(1). This crate was heavily influenced
-//! by the [LRU Cache implementation in an earlier version of Rust's std::collections crate](https://doc.rust-lang.org/0.12.0/std/collections/lru_cache/struct.LruCache.html).
+//! A simple, safe LRU cache implementation.
```

Comment:
I think this needs a Copyright notice as well

Reply:
Added